lib/common/repository/db_repository.dart

```diff
@@ -16,6 +16,46 @@ import 'package:path_provider/path_provider.dart' as path_provider;
 
 @singleton
 class DBRepository {
+  Future<List<LocalSession>> searchSessions(String query) async {
+    return localDB.localSessions
+        .where()
+        .filter()
+        .titleContains(query, caseSensitive: false)
+        .or()
+        .descriptionContains(query, caseSensitive: false)
+        .findAll();
+  }
+
+  Future<List<LocalSpeaker>> searchSpeakers(String query) async {
+    return localDB.localSpeakers
+        .where()
+        .filter()
+        .nameContains(query, caseSensitive: false)
+        .or()
+        .biographyContains(query, caseSensitive: false)
+        .or()
+        .taglineContains(query, caseSensitive: false)
+        .findAll();
+  }
+
+  Future<List<LocalSponsor>> searchSponsors(String query) async {
+    return localDB.localSponsors
+        .where()
+        .filter()
+        .nameContains(query, caseSensitive: false)
+        .or()
+        .taglineContains(query, caseSensitive: false)
+        .findAll();
+  }
+
+  Future<List<LocalOrganiser>> searchOrganisers(String query) async {
+    return localDB.localOrganisers
+        .where()
+        .filter()
+        .nameContains(query, caseSensitive: false)
+        .findAll();
+  }
+
```

Comment:
Please move these functions to the end of this file so that the top remains as it was.

Reply:
done